noir-projects/noir-protocol-circuits/crates/types/src/utils/arrays.nr

```diff
@@ -104,7 +106,11 @@ pub fn array_length<T, let N: u32>(array: [T; N]) -> u32
 where
     T: Empty + Eq,
 {
-    let length = unsafe { find_index_hint(array, |elem: T| is_empty(elem)) };
+    // We get the length by checking the index of the first empty element.
+    let length = unsafe {
+        //@safety This is safe because we have validated the array and the element emptyness is checked below.
+        find_index_hint(array, |elem: T| is_empty(elem))
+    };
```

Comment:
This is incomplete: we're only checking that the element at `length` is empty and that the previous element is non-empty, but we're not checking that all elements before `length` are non-empty. This is I think what the comment above re. requiring `validate_array` returning true alludes to. We should mention this.